src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/Invitations/InvitationsControllerNegativeTests.cs

```diff
@@ -598,6 +598,88 @@ public async Task AcceptPunchOut_AsAccepter_ShouldReturnBadRequest_WhenUnknownIn
         }
         #endregion
 
+        #region UnAccept
+        [TestMethod]
+        public async Task UnAcceptPunchOut_AsAnonymous_ShouldReturnUnauthorized()
+            => await InvitationsControllerTestsHelper.UnAcceptPunchOutAsync(
+                UserType.Anonymous,
+                TestFactory.PlantWithoutAccess,
+                9999,
+                new UnAcceptPunchOutDto(),
+                HttpStatusCode.Unauthorized);
+
+        [TestMethod]
+        public async Task UnAcceptPunchOut_AsHacker_ShouldReturnBadRequest_WhenUnknownPlant()
+            => await InvitationsControllerTestsHelper.UnAcceptPunchOutAsync(
+                UserType.Hacker,
+                TestFactory.UnknownPlant,
+                9999,
+                new UnAcceptPunchOutDto(),
+                HttpStatusCode.BadRequest,
+                "is not a valid plant");
+
+        [TestMethod]
+        public async Task UnAcceptPunchOut_AsAccepter_ShouldReturnBadRequest_WhenUnknownPlant()
+            => await InvitationsControllerTestsHelper.UnAcceptPunchOutAsync(
+                UserType.Accepter,
+                TestFactory.UnknownPlant,
+                9999,
+                new UnAcceptPunchOutDto(),
+                HttpStatusCode.BadRequest,
+                "is not a valid plant");
+
+        [TestMethod]
+        public async Task UnAcceptPunchOut_AsHacker_ShouldReturnForbidden_WhenPermissionMissing()
+            => await InvitationsControllerTestsHelper.UnAcceptPunchOutAsync(
+                UserType.Hacker,
+                TestFactory.PlantWithAccess,
+                9999,
+                new UnAcceptPunchOutDto(),
+                HttpStatusCode.Forbidden);
+
+        [TestMethod]
+        public async Task UnAcceptPunchOut_AsPlanner_ShouldReturnForbidden_WhenPermissionMissing()
+            => await InvitationsControllerTestsHelper.UnAcceptPunchOutAsync(
+                UserType.Planner,
+                TestFactory.PlantWithAccess,
+                9999,
+                new UnAcceptPunchOutDto(),
+                HttpStatusCode.Forbidden);
+
+        [TestMethod]
+        public async Task UnAcceptPunchOut_AsSigner_ShouldReturnForbidden_WhenPermissionMissing()
+            => await InvitationsControllerTestsHelper.UnAcceptPunchOutAsync(
+                UserType.Signer,
+                TestFactory.PlantWithAccess,
+                9999,
+                new UnAcceptPunchOutDto(),
+                HttpStatusCode.Forbidden);
+
+        [TestMethod]
+        public async Task UnAcceptPunchOut_AsAccepter_ShouldReturnBadRequest_WhenUnknownInvitationId()
+        {
+            var validInvitation = await InvitationsControllerTestsHelper.GetInvitationAsync(
```

Comment:
why find a validInvitation to test UnknownInvitationId?
Remove line 661 to 665, and just use an empty UnAcceptPunchOutDto() here too? (I tried and still a green test)